index.js

```diff
@@ -8,11 +8,100 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
+
+/**
+ * Check if the given `filepath` exists.
+ *
+ * ```js
+ * var res = exists('package.json');
+ * console.log(res);
+ * //=> "package.json"
+ *
+ * var res = exists('fake-file.json');
+ * console.log(res)
+ * //=> false
+ * ```
+ *
+ * @name exists
+ * @param  {String} `filepath` filepath to check for.
+ * @return {String|Boolean} Returns the found filepath if it exists, otherwise returns `false`.
+ * @api public
+ */
 
 module.exports = function(filepath) {
+  if (!filepath || (typeof filepath !== 'string')) {
+    return false;
+  }
+
   try {
     (fs.accessSync || fs.statSync)(filepath);
-    return true;
+    return filepath;
   } catch (err) {}
+
+  if (process.platform === 'linux') {
```

Comment:
I see what you mean about other file systems. Per my additional comments, I think a good solution is to just make the check optional, then we can keep this a drop-in replacement and allow the user to decide if the new logic will be used. 

Regardless of whether or not the logic is done here, I think this should be optional for the user. I personally would avoid using it if it was turned on by default, because it would be slow and I don't use a file system that needs the check.